File: 1._A._Latihan_Random_Coding/A._Google_Colab_Python/Latihan_Teknik_Perulangan_Fungsi_Modular.py
# Teknik Perulangan Fungsi Modular

print ("\n Teknik Perulangan Fungsi Modular \n")


print ("\n Perulangan dasar 1 \n")

for a in range (1, 11):
     print (f"Urutan ke - {a}")
     
print ("\n --- Batas --- \n")




print ("\n Perulangan dasar 2 \n")

for b in range (10, 21):
     print (f"Urutan ke - {b}")
   
     
print ("\n --- Batas --- \n")




print ("\n For Perulangan \n")

for c in range (15):
     print (f"Urutan ke - {c}")
     

print ("\n --- Batas --- \n")




print ("\n While Perulangan \n")

a = 10

while a < 25:
     print (f"Urutan ke - {a}")
     a = a + 1
     
     
print ("\n --- Batas --- \n")




print ("\n While Perulangan \n")

b = 15

while b < 25:
     print (f"Urutan ke - {b}")
     b = b + 1
     
     
print ("\n --- Batas --- \n")



print ("\n While Perulangan 3 \n")

c = 15

while c < 26:
     print (f"urutan ke - {c}")
     c = c + 1
     
     
print ("\n --- Batas --- \n")    



print ("\n While Perulangan 4 \n")

d = 10

while d > 0:
     print (f"Hitung mundur, angka = {d}")
     d = d - 1

print ("\n --- Batas --- \n")  




print ("\n For Nested \n")

for a in range (6):
     for b in range (6):
          print (f"Luar : {a}, Dalam : {b}")
          
          
print ("\n --- Batas --- \n")




print ("\n Teknik Iterasi For (Oshi saya) \n")

oshi = [
    "1. Michie JKT48 (UTAMA)",
    "2. Gracie JKT48 (UTAMA)",
    "3. Fritzy JKT48 (UTAMA)",
    "4. Lily JKT48 (UTAMA)",
    "5. Anindya JKT48 (UTAMA)",
    "6. Christy JKT48 (UTAMA)",
    "7. Freya JKT48 (UTAMA)",
    "8. Olla JKT48",
    "9. Jessi JKT48",
    "10. Fiony JKT48",
    "11. Muthe JKT48",
    "12. Marsha JKT48",
    "13. Eli JKT48",
    "14. Mikaela JKT48",
    "15. Ekin JKT48",
]

for a in oshi:
     print (a)


print ("\n --- Batas --- \n")




print ("\n Teknik Iterasi Kontrol Continue \n")

for f in range (1, 15):
     if f == 5:
          continue
     print (f"Urutan ke - {f}")
     
     
print ("\n --- Batas --- \n")




print ("\n Teknik Iterasi Kontrol Break\n")

for o in range (1, 25):
     if o == 15:
          break
     print (f"Urutan ke - {o}")
     
     
print ("\n --- Batas --- \n")




print ("\n List Iterasi Kontrol Continue \n")

buah = [
     "Apel", 
     "Naga", 
     "Melon", 
     "Semangka", 
     "Nangka",
     "Jeruk",
     "Strawberi",
     "Salak",
     
     ]

for a in buah:
     if a == "Melon":
          continue
     print (a)
     
     
print ("\n --- Batas --- \n")




print ("\n List Iterasi Kontrol Break \n")

buah = [
     "Apel", 
     "Naga", 
     "Melon", 
     "Semangka", 
     "Nangka",
     "Jeruk",
     "Strawberi",
     "Salak",
     
     ]

for k in buah:
     if k == "Melon":
          break
     print (k)
     
     
print ("\n --- Batas --- \n")




print ("\n Fungsi Dasar \n")

def tan ():
     print ("Hello World")
     
tan ()


print ("\n --- Batas --- \n")




print ("\n Fungsi dengan parameter \n")

def yun (nama):
     print (f"Halo nama saya {nama}, dari Jakarta Utara")
     
yun ("Fakhri")
yun ("Hayyan")
yun ("Rayyan")
yun ("Dimas")
yun ("Mido")


print ("\n --- Batas --- \n")



print ("\n Fungsi dengan parameter 1 \n")

def tur (nama, asal, suku, budaya):
     print (f"Halo nama saya {nama}, asal dari {asal}, suku saya adalah {suku}, dan budaya saya adalah {budaya}")
     
tur ("Frederick", "Jerman", "Jermanik Barat", "Jerman")
tur ("Louis", "Prancis", "Prancis", "Prancis")
tur ("Graciantsya", "Palembang", "Tionghoa", "Tionghoa")
tur ("Eri Erria", "Singkawang", "Tionghoa", "Tionghoa")


print ("\n --- Batas --- \n")



print ("\n Fungsi dengan Parameter \n")

def run (nama):
     print (f"Halo saya {nama} dari Jakarta")
     
run ("Rummer")
run ("Santer")
run ("Rus")
run ("Far")


print ("\n --- Batas --- \n")



print ("\n Fungsi dengan Return \n")

def tambah (a, b):
     return a + b

hasil = tambah (10, 10)
print (hasil)


print ("\n --- Batas --- \n")

File: 1._A._Latihan_Random_Coding/A._Online_JavaScript_Latihan/Latihan_Teknik_Perulangan_Fungsi_Modular.js
// Teknik Perulangan Fungsi Modular

console.log ("\n Teknik Perulangan Fungsi Modular \n")

for (a = 1; a < 11; a++) {
     console.log (`Urutan ke - ${a}`)
}


console.log ("\n --- Batas --- \n")




console.log ("\n For Perulangan 2 \n")

for (b = 10; b < 21; b++) {
     console.log (`Urutan ke - ${b}`)
}

console.log ("\n --- Batas --- \n")




console.log ("\n For Perulangan 3 \n")

for (c = 0; c < 15; c++) {
     console.log (`Urutan ke - ${c}`)
}


console.log ("\n --- Batas --- \n")




console.log ("\n While Perulangan 1 \n")

var a = 10

while (a < 20) {
     console.log (`Urutan ke - ${a}`)
     a++
}

console.log ("\n --- Batas --- \n")




console.log ("\n While Perulangan 2 \n")

var b = 15

while (b < 26) {
     console.log (`Urutan ke - ${b}`)
     b++
}


console.log ("\n --- Batas --- \n")




console.log ("\n While Perulangan 3 \n")

var c = 10

while (c < 26) {
     console.log (`Urutan ke - ${c}`)
     c++
}


console.log ("\n --- Batas --- \n")



console.log ("\n While Perulangan 4 \n")

var d = 10

while (d > 0) {
     console.log (`Hitung mundur, d = ${d}`)
     d--
}


console.log ("\n --- Batas --- \n")




console.log ("\n For Nested \n")

for (a = 0; a < 6; a++) {
     for (b = 0; b < 6; b++) {
          console.log (`Luar : ${a}, Dalam : ${b}`)
     }
}


console.log ("\n --- Batas --- \n")




console.log ("\n Teknik Iterasi For (Oshi saya) \n")

oshi = [
    "1. Michie JKT48 (UTAMA)",
    "2. Gracie JKT48 (UTAMA)",
    "3. Fritzy JKT48 (UTAMA)",
    "4. Lily JKT48 (UTAMA)",
    "5. Anindya JKT48 (UTAMA)",
    "6. Christy JKT48 (UTAMA)",
    "7. Freya JKT48 (UTAMA)",
    "8. Olla JKT48",
    "9. Jessi JKT48",
    "10. Fiony JKT48",
    "11. Muthe JKT48",
    "12. Marsha JKT48",
    "13. Eli JKT48",
    "14. Mikaela JKT48",
    "15. Ekin JKT48",
]

for (a = 0; a < oshi.length; a++) {
     console.log (oshi [a])
}


console.log ("\n --- Batas --- \n")




console.log ("\n Teknik Iterasi Kontrol Continue \n")

for (a = 1; a < 15; a++) {
     if (a == 5) {
          continue
     }

     console.log (`Urutan ke - ${a}`)
}

console.log ("\n --- Batas --- \n")



console.log ("\n Teknik Iterasi Kontrol Break \n")

for (b = 1; b < 25; b++) {
     if (b == 10) {
          break
     }

     console.log (`Urutan ke - ${b}`)
}


console.log ("\n --- Batas --- \n")




console.log ("\n List Iterasi Kontrol Continue \n")

var buah = [
     "Apel", 
     "Naga", 
     "Melon", 
     "Semangka", 
     "Nangka",
     "Jeruk",
     "Strawberi",
     "Salak",
     
     ]

for (var a of buah) {
     if (a == "Melon") {
          continue
     }

     console.log (a)
}


console.log ("\n --- Batas --- \n")




console.log ("\n List Iterasi Kontrol Break \n")

var buah = [
     "Apel", 
     "Naga", 
     "Melon", 
     "Semangka", 
     "Nangka",
     "Jeruk",
     "Strawberi",
     "Salak",
     
     ]

for (var b of buah) {
     if (b == "Melon") {
          break
     }

     console.log (b)
} 


console.log ("\n --- Batas --- \n")




console.log ("\n Fungsi dasar \n")

function tan () {
     console.log ("Hello World")
}

tan ()


console.log ("\n --- Batas --- \n")




console.log ("\n Fungsi dengan parameter 1 \n")

function yun (nama) {
     console.log (`Halo nama saya ${nama}, dari Jakarta Utara`)
}

yun ("Fakhri")
yun ("Hayyan")
yun ("Rayyan")
yun ("Dimas")
yun ("Mido")


console.log ("\n --- Batas --- \n")




console.log ("\n Fungsi dengan Parameter 2 \n")

function tur (nama, asal, suku, budaya) {
     console.log (`Halo nama saya ${nama}, asal dari ${asal}, suku saya adalah ${suku}, dan budaya saya adalah ${budaya}`)
}

tur ("Frederick", "Jerman", "Jermanik Barat", "Jerman")
tur ("Louis", "Prancis", "Prancis", "Prancis")
tur ("Graciantsya", "Palembang", "Tionghoa", "Tionghoa")
tur ("Eri Erria", "Singkawang", "Tionghoa", "Tionghoa")


console.log ("\n --- Batas --- \n")



console.log ("\n Fungsi dengan Parameter \n")

function run (nama) {
     console.log (`Halo saya ${nama} dari Jakarta`)
}

run ("Rummer")
run ("Santer")
run ("Rus")
run ("Far")


console.log ("\n --- Batas --- \n")




console.log ("\n Fungsi dengan Return \n")

function tambah (x, y) {
     return x + y
}

hasil = tambah (10, 10)
console.log ("Tambah =", hasil)

console.log ("\n --- Batas --- \n")